pkg/controller/perconaservermongodb/status.go

```diff
@@ -212,17 +213,20 @@ func (r *ReconcilePerconaServerMongoDB) upgradeInProgress(cr *api.PerconaServerM
 }
 
 func (r *ReconcilePerconaServerMongoDB) writeStatus(cr *api.PerconaServerMongoDB) error {
-	err := r.client.Status().Update(context.TODO(), cr)
-	if err != nil {
-		// may be it's k8s v1.10 and erlier (e.g. oc3.9) that doesn't support status updates
-		// so try to update whole CR
-		err := r.client.Update(context.TODO(), cr)
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		c := &api.PerconaServerMongoDB{}
+
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, c)
 		if err != nil {
-			return errors.Wrap(err, "send update")
+			return err
```

Comment:
I think we need wrapper here

Reply:
we shouldn't wrap this to `retryOnConflict` to work: https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict